pkg/dynamicinstrumentation/codegen/codegen.go

```diff
@@ -223,10 +239,81 @@ func generateSliceHeader(slice *ditypes.Parameter, out io.Writer) error {
 	if err != nil {
 		return fmt.Errorf("could not execute template for generating slice header: %w", err)
 	}
+
+	return nil
+}
+
+func generateStringHeader(stringParam *ditypes.Parameter, out io.Writer) error {
+	if stringParam == nil {
+		return errors.New("nil string parameter when generating header code")
+	}
+	if len(stringParam.ParameterPieces) != 2 {
+		return fmt.Errorf("invalid string parameter when generating header code (pieces len %d)", len(stringParam.ParameterPieces))
+	}
+
+	x := []byte{}
+	buf := bytes.NewBuffer(x)
+	err := generateStringLengthHeader(stringParam.ParameterPieces[1], buf)
+	if err != nil {
+		return err
+	}
+
+	stringHeaderWrapper := stringHeaderWrapper{
+		Parameter:        stringParam,
+		StringLengthText: buf.String(),
+	}
```

Comment:
What do we gain by separating the length template from the main header one and injecting the length as text?

Reply:
Precisely reading the length directly from where it's passed. What we were doing before was looking at what register the `ptr` was passed in, and adding 1 to it. This works on ARM but not x86 (the x86 ABI is strange).